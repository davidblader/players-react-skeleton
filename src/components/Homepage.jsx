import React from 'react';
import { Link } from 'react-router-dom';
import AnimalCrossingHeader from './AnimalCrossingHeader';
import Button from './Button';
import AnimalCrossingContainer from './AnimalCrossingContainer';
import Image from './Image';

const Homepage = () =>
  (
    <AnimalCrossingContainer>
      <AnimalCrossingHeader>Animal Crossing</AnimalCrossingHeader>
      <AnimalCrossingHeader>Fantasy Village</AnimalCrossingHeader>
      <Image src="villager" />
      <div className="btn-container">
        <Link id="login-link" to="/login" href="/login">
          <Button id="login-btn">Login</Button>
        </Link>
      </div>
      <div className="btn-container">
        <Link id="register-link" to="/register" href="/register">
          <Button id="register-btn">Register</Button>
        </Link>
      </div>
    </AnimalCrossingContainer>
  );

export default Homepage;
