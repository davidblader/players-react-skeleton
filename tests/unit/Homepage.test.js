import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Homepage from '../../src/components/Homepage';

test('Homepage shallow renders', () => {
  const component = shallow(<Homepage />);
  expect(component).toMatchSnapshot();
});

test('Login link leads to /login', () => {
  const component = mount(<MemoryRouter><Homepage /></MemoryRouter>);
  expect(component.find('Link#login-link').prop('to')).toBe('/login');
  component.unmount();
});

test('Register link leads to /register', () => {
  const component = mount(<MemoryRouter><Homepage /></MemoryRouter>);
  expect(component.find('Link#register-link').prop('to')).toBe('/register');
  component.unmount();
});
