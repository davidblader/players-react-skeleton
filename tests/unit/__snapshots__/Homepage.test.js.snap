// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage shallow renders 1`] = `
<AnimalCrossingContainer>
  <AnimalCrossingHeader>
    Animal Crossing
  </AnimalCrossingHeader>
  <AnimalCrossingHeader>
    Fantasy Village
  </AnimalCrossingHeader>
  <Image
    src="villager"
  />
  <div
    className="btn-container"
  >
    <Link
      href="/login"
      id="login-link"
      replace={false}
      to="/login"
    >
      <Button
        id="login-btn"
      >
        Login
      </Button>
    </Link>
  </div>
  <div
    className="btn-container"
  >
    <Link
      href="/register"
      id="register-link"
      replace={false}
      to="/register"
    >
      <Button
        id="register-btn"
      >
        Register
      </Button>
    </Link>
  </div>
</AnimalCrossingContainer>
`;
